src/utils/parseCSV.ts

```diff
@@ -0,0 +1,27 @@
+import TestCSV from '@/data/sample.csv?raw';
+
+/**
+ * Parses a CSV string from a local file and returns an array of rows.
+ * Sanitizes cell data to prevent injection attacks.
+ * @param csvString The raw CSV string content to parse.
+ * @returns The parsed CSV data as a 2D array of strings.
+ */
+export const parseCSV = (csvString: string = TestCSV): string[][] => {
+  try {
+    const rows = csvString.split('\n');
+    return rows.map((row) => row.split(',').map((cell) => sanitizeCell(cell)));
+  } catch (error) {
+    console.error('Error parsing CSV:', error);
+    throw new Error('Failed to parse CSV file.');
+  }
+};
+
+/**
+ * Sanitizes cell content to prevent CSV injection and other potential vulnerabilities.
+ * Strips or encodes harmful characters that could be used in an attack.
+ * @param cell The cell content to sanitize.
+ * @returns The sanitized cell content.
+ */
+const sanitizeCell = (cell: string): string => {
+  return cell.trim().replace(/^[=+\-@]/, '');
```

Comment:
we'll probably want to do a bit more here - i feel like simply stripping out the characters could be confusing

Reply:
> Alternatively, apply the following sanitization to each field of the CSV, so that their content will be read as text by the spreadsheet editor:
> * Wrap each cell field in double quotes
> * Prepend each cell field with a single quote
> * Escape every double quote using an additional double quote

possibly just these items ref: https://owasp.org/www-community/attacks/CSV_Injection